Tests/SCM.SwissArmyKnife.Test/GzipTests/GzipTests.cs

```diff
@@ -0,0 +1,183 @@
+
+            //Assert
+            compressedString.Should().BeOfType(typeof(byte[]));
+        }
+
+        [Fact]
+        public void Compress_WithEmptyStringAndNoEncoding_OutputShouldBeByteArray()
+        {
+            //Arrange
+            var emptyString = string.Empty;
+
+            //Act
+            var compressedString = Gzip.Compress(emptyString);
+
+            //Assert
+            compressedString.Should().BeOfType(typeof(byte[]));
+        }
+
+        [Fact]
+        public void Compress_WithEmptyStringAndWithEncoding_OutputShouldBeByteArray()
+        {
+            //Arrange
+            var encoding = Encoding.ASCII;
+            var emptyString = string.Empty;
+
+            //Act
+            var compressedString = Gzip.Compress(emptyString, encoding);
+
+            //Assert
+            compressedString.Should().BeOfType(typeof(byte[]));
+        }
+
+
+        [Fact]
+        public void Decompress_OutputShouldBeEmpty_IfInputEmpty()
+        {
+            // Arrange
+            var testData = Array.Empty<byte>();
+
+            // Act
+            var decompressedData = Gzip.Decompress(testData);
+
+            // Assert
+            decompressedData.Should().BeEmpty();
+        }
+
+        [Fact]
+        public void Decompress_ShoudCreateOutput_ThatIsLargerThanTheInput()
+        {
+            // Arrange
+            var testData = File.ReadAllBytes(@$"GzipTests{Path.DirectorySeparatorChar}TestingFiles{Path.DirectorySeparatorChar}TestCompressedData.gz");
+
+            // Act
+            var decompressedData = Gzip.Decompress(testData);
+
+            // Assert
+            decompressedData.Length.Should().BeGreaterThan(testData.Length);
+        }
+
+        [Fact]
+        public void Decompress_OutputShouldMatchControlText()
+        {
+            // Arrange
+            var testData = File.ReadAllBytes(@$"GzipTests{Path.DirectorySeparatorChar}TestingFiles{Path.DirectorySeparatorChar}TestCompressedData.gz");
+
+            // Act
+            var decompressedData = Gzip.Decompress(testData);
+            var decompressedString = Encoding.ASCII.GetString(decompressedData);
+
+            // Assert
+            decompressedString.Should().BeEquivalentTo(testString);
+        }
+
+
+        [Fact]
+        public void DecompressToString_OutputShouldMatchControlText()
+        {
+            // Arrange
+            var testData = File.ReadAllBytes(@$"GzipTests{Path.DirectorySeparatorChar}TestingFiles{Path.DirectorySeparatorChar}TestCompressedData.gz");
+            var encoding = Encoding.ASCII;
+
+            // Act
+            var decompressedString = Gzip.DecompressToString(testData, encoding);
+
+            // Assert
+            decompressedString.Should().BeEquivalentTo(testString);
+        }
+
+        [Fact]
+        public void CompressDecompress_OutputShouldMatchControlString()
+        {
+            //Not the most reliable test, but for what it is worth I think it still brings some value.
```

Comment:
I agree.

Reply:
I suggest you make another test like this for the `String` overloads, and then axe all the tests above with the `compressedString.Should().BeOfType(typeof(byte[]));` assertions.